File: haskell/Main4.hs
dangerous :: Int -> Int -> Int
dangerous v1 v2 = v1 `div` v2
foo :: [Int] -> Int -> Int
foo array counter = if counter == 0 then dangerous (array !! 0) counter else foo array (counter - 1)
main :: IO ()
main = do
  let array = replicate 1000 0
  print (foo array 6)

File: nim/main4.nim
proc dangerous(v1:int, v2:int): int = v1 div v2
proc foo(array: seq[int], counter:int): int =
  if counter == 0: dangerous(array[0], counter)
  else: foo(array, counter - 1)
when isMainModule:
  var array = newSeq[int](1000)
  echo "The result is ", foo(array, 6)

